File: Two-Sum-II-Input-Array-Is-Sorted/Two-Sum-II-Input-Array-Is-Sorted.js
let numbers = [2,3,4], target = 6

var twoSum = function(numbers, target) {
    
    let i = 0, j=numbers.length-1
    let myMap = new Map();
    // dùng two point
    while(i<j){
        if(numbers[i]+numbers[j]===target){
            return [i+1,j+1]
        }
        if(numbers[i]+numbers[j]>target){
            j--
            i=0
        }
        if(numbers[i]+numbers[j]<target){
            i++
        }
    }

    // dùng map()
    // for(let i = 0;i<=numbers.length-1;i++){
    //     if(myMap.has(numbers[i])){
    //         return [Number(myMap.get(numbers[i]))+1,i+1]
    //     }else{
    //         let a = target - numbers[i]
    //         myMap.set(a,i)
    //     }
    // }

};

console.log(twoSum(numbers,target))

File: Two-Sum-II-Input-Array-Is-Sorted/Two-Sum-II-Input-Array-Is-Sorted.py
numbers = [2,7,11,15]
target = 9

def twoSum(numbers,target):
    i = 0 
    j=len(numbers)-1
    while i<j:
        if numbers[i]+numbers[j]==target:
            return [i+1,j+1]
        if numbers[i]+numbers[j]>target:
            j=j-1
            i=0
        if numbers[i]+numbers[j]<target:
            i=i+1

print(twoSum(numbers,target))